rendermodules/intensity_correction/apply_multiplicative_correction.py

```diff
@@ -127,9 +156,31 @@ def run(self):
         corr_tilespecs = renderapi.tilespec.get_tile_specs_from_z(
             self.args['correction_stack'], Z, render=self.render)
         # mult intensity correct each tilespecs and return tilespecs
-        N, M, C = getImage(corr_tilespecs[0])
+        corr_ts = corr_tilespecs[0]
+
+        N, M, C = getImage(corr_ts)
+
+        # construct a dictionary with the correction factors
+        corr_dict = {}
+        if corr_ts.channels is not None:
+            for chan in corr_ts.channels:
+                Nc, Mc, CC = getImage(corr_ts, chan.name)
+                corr_dict[chan.name] = CC
+        
+        outdir = self.args['output_directory']
+        if not os.path.exists(outdir):
```

Comment:
making output_directory an `argschema.fields.OutputDir` in the schema will create this while also checking for directory mode and avoiding race conditions.

Reply:
Thanks for the pointer.